crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -426,34 +474,47 @@ struct UpstreamCatalog<'a>(&'a Git);
 impl UpstreamCatalog<'_> {
     /// Mostly naïve approxiaton of a snapshot path
     ///
-    ///  /packages/<pname>/<version>.json
+    ///  /catalog/<namespace '/'-separated>/<version>-<(filename outPaths[0])[..8]>.json
     ///
-    /// TODO: fix before releasing publish!
-    fn get_snapshot_path(&self, snapshot: &Value) -> PathBuf {
-        let path = self
-            .0
-            .workdir()
+    /// see [tests::test_get_snapshot_path] for an example
+    fn get_snapshot_path(snapshot: &Value) -> PathBuf {
+        let mut path = PathBuf::from("catalog");
+
+        let attr_path = snapshot["eval"]["attrPath"]
+            .as_array()
             .unwrap()
-            .join("packages")
-            .join(
-                snapshot["eval"]["meta"]["pname"]
-                    .as_str()
-                    .expect("'pname' is expected to be a string"),
-            )
-            .join(format!(
-                "{}.json",
-                snapshot["eval"]["meta"]["version"]
-                    .as_str()
-                    .expect("'version' is expected to be a string")
-            ));
+            .iter()
+            .skip(2)
+            .map(|v| v.as_str().unwrap());
+
+        path.extend(attr_path);
+
+        let version = &snapshot["eval"]["version"]
+            .as_str()
+            .expect("invalid metadata");
+
+        let nix_out_hash = &Path::new(
+            snapshot["element"]["storePaths"][0]
```

Comment:
Do we preserve order everywhere? Seems a bit unreliable to assume we will

Reply:
We shouldn't be modifying these files anyway, nix depends on the order of these i think and I dont know where to get a "default" output hash from otherwise

Reply:
Does nix even have `storePaths`? I think it might rely on the order of `outputs`? But I'm a bit skeptical that we just got lucky and happened to both preserve the order of outputs and storePaths. You could just hash the list of all store paths - that would have the additional benefit(?) of preventing collisions between `flox publish attr^out` and `flox publish attr^dev`